src/parser/shared/modules/spells/bfa/essences/BloodOfTheEnemy.js

```diff
@@ -0,0 +1,183 @@
+};
+
+/**
+ * Major: The Heart of Azeroth erupts violently, dealing 19,408 Shadow damage to enemies within 12 yds. You gain 25% critical strike chance against the targets for 10 sec.
+ * Minor: Your critical strikes with spells and abilities grant a stack of Blood-Soaked. Upon reaching 40 stacks, you gain 548 Haste for 8 sec.
+ * R2: 30 second CD reduction / Each stack of Blood-Soaked grants 8 Critical Strike
+ * R3: increases your critical hit damage by 25% for 15 sec / Blood-Soaked has a 25% chance of only consuming 30 stacks.
+ */
+class BloodOfTheEnemy extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+    statTracker: StatTracker,
+    enemies: EnemyInstances,
+  }
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasEssence(SPELLS.BLOOD_OF_THE_ENEMY.traitId);
+    if (!this.active) {
+      return;
+    }
+
+    const rank = this.selectedCombatant.essenceRank(SPELLS.BLOOD_OF_THE_ENEMY.traitId);
+    this.statTracker.add(SPELLS.BLOOD_SOAKED_HASTE_BUFF.id, {
+      crit: (rank === 1) ? 0 : 8,
+      haste: 548,
+    });
+
+    this.statTracker.add(SPELLS.BLOOD_SOAKED_STACKING_BUFF.id, {
+      crit: (rank === 1) ? 0 : 8,
+    });
+
+    
+    this.hasMajor = this.selectedCombatant.hasMajor(SPELLS.BLOOD_OF_THE_ENEMY.traitId);
+    if(this.hasMajor) {
+      this.abilities.add({
+        spell: SPELLS.BLOOD_OF_THE_ENEMY,
+        category: Abilities.SPELL_CATEGORIES.ITEMS,
+        cooldown: (rank === 1) ? 120 : 90,
+        gcd: {
+        base: 1500,
+        },
+        castEfficiency: {
+          suggestion: true,
+          recommendedEfficiency: 0.80,
+        },
+      });
+    }
+    
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.BLOOD_OF_THE_ENEMY), this.onMajorCastDamage);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDamage);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.BLOOD_SOAKED_STACKING_BUFF), this.handleStacks);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.BLOOD_SOAKED_STACKING_BUFF), this.handleStacks);
+    this.addEventListener(Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.BLOOD_SOAKED_STACKING_BUFF), this.handleStacks);
+  }
+
+  majorCastDamage = 0;
+  currentStacks = 0;
+  lastStackTimestamp = 0;
+  totalCrit = 0;
+  bonusDamage = 0;
+
+  onMajorCastDamage(event){
+    this.majorCastDamage += event.amount + (event.absorbed || 0);
+  }
+
+  onDamage(event){
+    const enemy = this.enemies.getEntity(event);
+    if (!enemy) {
+      return;
+    }
+    if (!enemy.hasBuff(SPELLS.BLOOD_OF_THE_ENEMY.id, event.timestamp, 0, 0, this.selectedCombatant)) {
+      return;
+    }
+    const isCrit = event.hitType === HIT_TYPES.CRIT || event.hitType === HIT_TYPES.BLOCKED_CRIT;
+    if (!isCrit) {
+      return;
+    }
+
+    this.bonusDamage += calculateEffectiveDamage(event, BOTE_DAMAGE_BONUS);
```

Comment:
the bonus crit damage is only at R3, but this event handler fires for R1 and R2 as well. should only add the event listener for R3 BotE